content/articles/create-screen-recorder-in-javascript/index.md

```diff
@@ -0,0 +1,219 @@
+
+### HTML Page
+On the `index.html` page there will be
+* Two buttons one to Start Screen Recording another to Stop Screen Recording.
+* One `div` to display the recording status, the style for the div is also added in the `index.html` file.
+
+```html
+<!DOCTYPE html>
+<html lang="en" dir="ltr">
+    <head>
+        <meta charset="utf-8">
+        <title>JavaScript Screen Recorder</title>
+        <style>
+            .status {
+                background: pink;
+                font-size: 30px;
+                padding: 10px;
+                margin : 0;
+                margin-bottom: 30px;
+                line-height: 40px;
+                box-sizing: border-box;
+            }
+        </style>
+    </head>
+    <body>
+        <div class="status">Record your screen by pressing Start Screen Recording </div>
+        <button id="start">Start Recording</button>
+        <button id="stop">Stop Recording</button>
+        <script src="recordscreen.js" charset="utf-8"></script>
+    </body>
+</html>
+```
+### The JavaScript file
+
+In the JavaScript file, we will have 3 methods
+1. `recordScreen` -- This will start recording the user's screens.
+2. `createMediaStream` -- This will create a MediaStream object from the screen stream.
+3. `saveRecording` -- This will save the recorded stream as a file to our computer.
+
+#### Record Screen
+
+On calling `getDisplayMedia` method in `navigator.mediaDevices` object will prompt the user to select and give permission to record the screen or part of the screen (browser tab). If the user has multiple screens then all the screens are displayed.
+
+For `getDisplayMedia`  method we need to pass an object([constraints](https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices/getDisplayMedia#parameters)) as argument  with two property `audio` and `video` . If we set the value of `audio` to true then the audio is included in the stream.  In our case, we need both audio and video.
+
+```js
+async function recordScreen() {
+	return await navigator.mediaDevices.getDisplayMedia({
+		audio: true,
+		video: true
+	});
+}
+```
+
+#### Create MediaRecorder
+
+The `getDisplayMedia` will return a stream data. From that, we need to create a `MediaRecorder`.
+
+ The`MediaRecorder` interface contains the following methods
+
+|Method   	  | Functionality                		 |
+|-------------|--------------------------------------|
+|start  	  |  Start recording the stream 		 |
+|pause  	  |  Pause the recording        		 |
+|resume 	  |  Resume the paused recording		 |
+|stop         |  Stop the recording			   		 |
+|requestData  |  Returns the recorded data as blob   |
+
+The important events available in `MediaRecorder` are
+
+**dataavailable**:
+
+This event is fired when the MediaRecorder delivers media data to your application for its use. We can collect the data from here and store it.
+
+**stop**:
+
+This will be triggered by calling the top method on the MediaRecorder interface.
+
+Steps to create a recorder
```

Comment:
I think you can have this at the beginning of this step.

Reply:
i have changed this